change top level heuristic PURL to guac purl

lumjjb opened the pull request:
Changing the top level heuristic from spdx document name to guac purl

pxp928 commented on pkg/ingestor/parser/spdx/parse_spdx.go:
```diff
@@ -76,9 +76,9 @@ func (s *spdxParser) getTopLevelPackage() error {
 	splitImage := strings.Split(s.spdxDoc.DocumentName, "/")
 	var purl string
 	if len(splitImage) == 3 {
```
can remove this check and `split` above as not needed.

mihaimaruseac replied:
+1

lumjjb replied:
apparently email replies to the entire issue, bringing the comments back to this thread 

>@lumjjb: We still need the check since it may not be an oci image
>@pxp928: Does it need to be type oci? Seems very error prone again splitting strings.

lumjjb replied:
well i think its a guac heuristic for OCI images, so maybe we can make it `guac/heuristic/oci/path`?

mihaimaruseac reviewed (approved):
LGTM, modulo Parth's question

lumjjb commented:
We still need the check since it may not be an oci image

On Tue, Mar 7, 2023, 3:18 PM Parth Patel ***@***.***> wrote:

> ***@***.**** approved this pull request.
> ------------------------------
>
> In pkg/ingestor/parser/spdx/parse_spdx.go
> <https://github.com/guacsec/guac/pull/544#discussion_r1128536788>:
>
> > @@ -76,9 +76,9 @@ func (s *spdxParser) getTopLevelPackage() error {
>  	splitImage := strings.Split(s.spdxDoc.DocumentName, "/")
>  	var purl string
>  	if len(splitImage) == 3 {
>
> can remove this check and split above as not needed.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/guacsec/guac/pull/544#pullrequestreview-1329461736>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAXLDBUFLPTQGHYYBAZVQFLW26J2HANCNFSM6AAAAAAVS5Y7EE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

pxp928 commented:
Does it need to be type `oci`? Seems very error prone again splitting strings.
